docs/docs-beta/docs/guides/ingestion-and-transformation/ingesting-data.md

```diff
@@ -0,0 +1,72 @@
+
+Dagster has a different relationship to streaming data ingestion than it does to batch data ingestion. For batch data ingestion, Dagster often takes responsibility for kicking off the computations to ingest the batches of data, either at a regular cadence or when it discovers that new data is available in the source system.
+
+For streaming data ingestion, Dagster doesn't orchestrate the data ingestion, but still often represents the ingested data assets in its asset graph. Dagster can hold metadata about these assets, represent the lineage between them and downstream assets, and automatically kick off computations to observe them.
+
+## Orchestrate a data ingestion tool
+
+Dagster integrates with several tools that are purpose-built for data ingestion. These tools roughly fall into two main categories:
+- Hosted data ingestion services.
+- Embeddable data ingestion libraries.
+
+### Hosted data ingestion services
+
+Dagster provides two integrations with two hosted data ingestion services: Fivetran and Airbyte.
+
+With a hosted data ingestion service, the set of data sources to ingest into the data platform is defined inside the service. Dagster's integrations with these services invoke their REST APIs to find out about the set of data assets that they ingest and load those asset definitions into Dagster's asset graph. The "materialize" action on these Dagster asset definitions invokes REST APIs provided by these services to ingest the latest data from the source.
+
+To learn how to use Dagster to orchestrate a hosted data ingestion service, follow the link for that service above.
+
+### Embeddable data ingestion libraries
+
+With an embeddable data ingestion library, the set of data sources to ingest into the data platform is defined using files that are typically managed inside the same git repository as other Dagster pipelines. Dagster's integrations with these libraries invokes them to interpret these files and load the data assets defined in them into Dagster's asset graph. The "materialize" action on these Dagster asset definitions invokes the library to ingest the latest data from the source.
+
+Dagster provides two integrations with embeddable data ingestion libraries: Sling and DLT.
+
+To learn how to use Dagster to orchestrate a hosted data ingestion service, follow the link for that library above.
+
+## Write a custom data ingestion pipeline
+
+It's also common to write code in a language like Python to ingest data into a data platform.
+
+For example, if there's a CSV file on the internet of counties, and you want to load it into your Snowflake data warehouse as a table, you might directly define an asset that represents that table in your warehouse. The asset's materialization function fetches data from the internet and loads it into that table.
+
+```python
```

Comment:
This example should be self-contained and fully runnable. 
Concept/Explanation pages don't have this constraint however, so using code to explain concepts is totally fine in those sections.

Reply:
Yeah this was just a placeholder. Do we have a general strategy for code examples where an external service is involved? I.e. the most realistic thing to put here would be something that downloads a file from S3 and writes it to a table in Snowflake, but most readers aren't going to have an S3 bucket with that particular file and a Snowflake instance.

Reply:
Pedram can weigh in, but I think that's fine. It should just be runnable for those that do have those things.

Reply:
I think it's fine for the file not to exist, expecting a user to replace a path before an example runs is totally fine when dealing with cloud service examples.